main.js

```diff
@@ -44,13 +45,23 @@ function parseInput() {
   }
 }
 
-async function downloadInstaller(destination) {
-  await tc.downloadTool(inst_url, destination);
+async function downloadInstaller() {
+  let inst_path = tc.find('msys2-installer', inst_version, 'x64');
+  let destination;
+  if (inst_path) {
+    destination = path.join(inst_path, 'base.exe');
+  } else {
+    destination = await tc.downloadTool(inst_url);
+  }
   let computedChecksum = '';
   await exec.exec(`powershell.exe`, [`(Get-FileHash ${destination} -Algorithm SHA256)[0].Hash`], {listeners: {stdout: (data) => { computedChecksum += data.toString(); }}});
   if (computedChecksum.slice(0, -2).toUpperCase() !== checksum.toUpperCase()) {
     throw new Error(`The SHA256 of the installer does not match! expected ${checksum} got ${computedChecksum}`);
   }
+  if (!inst_path) {
+    inst_path = await tc.cacheFile(destination, 'base.exe', 'msys2-installer', inst_version, 'x64');
+  }
+  return path.join(inst_path, 'base.exe');
```

Comment:
Similarly:

```js
return inst_path ? destination : await tc.cacheFile(destination, 'base.exe', 'msys2-installer', inst_version, 'x64');
```

Reply:
unfortunately tc.cacheFile returns the directory, not the file, so it still needs to be joined to 'base.exe'.  But I get the concept if you prefer ternaries I can do that.

Reply:
I overlooked that. In that case, it's ok to keep the if separated.

I do like ternaries when the solution is a single row (say 120-125 char max). Otherwise, it's confusing. The other suggestion was very obvious, but this was me overthinking.

Reply:
better now?